.github/workflows/tests.yml

```diff
@@ -58,10 +58,28 @@ jobs:
         run: |
           go mod download
 
+      - name: Make LXD tarball and unpack it
+        env:
+          CUSTOM_VERSION: "test"
+        run: |
+          make dist
+          tar -xzf lxd-test.tar.gz
+
+      - name: Build LXD dependencies
+        run: |
+          cd lxd-test
+          make deps
+
       - name: Run LXD build
         run: |
+          cd lxd-test
           make
```

Comment:
@MusicDin this `make` command isn't using the dependencies built in `make deps` above because the last output of `make deps` instructs to set these env vars, which isn't being done:

```
export CGO_CFLAGS="-I/home/runner/work/lxd/lxd/lxd-test/vendor/raft/include/ -I/home/runner/work/lxd/lxd/lxd-test/vendor/dqlite/include/"
export CGO_LDFLAGS="-L/home/runner/work/lxd/lxd/lxd-test/vendor/raft/.libs -L/home/runner/work/lxd/lxd/lxd-test/vendor/dqlite/.libs/"
export LD_LIBRARY_PATH="/home/runner/work/lxd/lxd/lxd-test/vendor/raft/.libs/:/home/runner/work/lxd/lxd/lxd-test/vendor/dqlite/.libs/"
export CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
```

This means that the `make` step is only succeeding due to using the dqlite ppa installed earlier:

```
sudo add-apt-repository ppa:dqlite/dev -y --no-update
```

We should remove that apt repository for the Code Tests test and not install `libdqlite-dev`, so that we actually test building raft and dqlite from the bundled tarball. This will require setting those env vars before calling `make`.